mobile/native/src/db/mod.rs

```
@@ -0,0 +1,58 @@
+pub mod models;
+
+use crate::db::models::LastLogin;
+use anyhow::anyhow;
+use anyhow::Context;
+use anyhow::Result;
+use diesel::r2d2;
+use diesel::r2d2::ConnectionManager;
+use diesel::r2d2::Pool;
+use diesel::r2d2::PooledConnection;
+use diesel::SqliteConnection;
+use diesel_migrations::embed_migrations;
+use diesel_migrations::EmbeddedMigrations;
+use diesel_migrations::MigrationHarness;
+use state::Storage;
+use std::sync::Arc;
+use time::OffsetDateTime;
+
+pub const MIGRATIONS: EmbeddedMigrations = embed_migrations!();
+
+/// Wallet has to be managed by Rust as generics are not support by frb
+static DB: Storage<Arc<Pool<ConnectionManager<SqliteConnection>>>> = Storage::new();
+
+pub fn init_db(database_url: String) -> Result<()> {
+    let manager = ConnectionManager::<SqliteConnection>::new(database_url);
+    let pool = r2d2::Pool::builder()
+        .build(manager)
+        .expect("Failed to create pool.");
```

❓ Why not bubble up the error since we are already returning a `Result`?